src/main/java/org/folio/inventory/dataimport/util/AdditionalFieldsUtil.java

```diff
@@ -523,7 +577,44 @@ && isNotEmpty(controlField.getData())
     return false;
   }
 
-  private static String getRecordId(Record record) {
-    return record != null ? record.getId() : "";
+  private static String getRecordId(Record srcRecord) {
+    return srcRecord != null ? srcRecord.getId() : "";
+  }
+
+  public static void remove035FieldWhenRecordContainsHrId(Record srcRecord) {
+    if (Record.RecordType.MARC_BIB.equals(srcRecord.getRecordType())) {
+      String hrid = getValueFromControlledField(srcRecord, TAG_001);
+      remove035WithActualHrId(srcRecord, hrid);
+    }
+  }
+
+  public static void remove035WithActualHrId(Record srcRecord, String actualHrId) {
+    removeField(srcRecord, TAG_035, TAG_035_SUB, actualHrId);
+  }
+
+  /**
+   * Move original marc hrId to 035 tag and assign created by inventory hrId into 001 tag
+   *
+   * @param recordInstancePair pair of related instance and record
+   */
+  public static void fillHrIdFieldInMarcRecord(Pair<Record, JsonObject> recordInstancePair) {
```

Comment:
this logic was copied from [srs](https://github.com/folio-org/mod-source-record-storage/blob/46f0abd66ad9f24c4c016834728b5cffda8049e6/mod-source-record-storage-server/src/main/java/org/folio/services/util/AdditionalFieldsUtil.java#L437). Also please consider review this method with [move001To035](https://github.com/folio-org/mod-inventory/blob/755f71ef1121ed926e37e2008c00ffd2d346f0d5/src/main/java/org/folio/inventory/dataimport/util/AdditionalFieldsUtil.java#L243)